src/handlers/VehicleTestController.ts

```diff
@@ -383,14 +387,14 @@ export class VehicleTestController implements IVehicleTestController {
   }
 
   public async checkTestTypeStartAndEndTimestamp(
-    newTestResult: models.ITestResult,
+    newTestResult: TestResultSchema,
   ) {
     const { testTypes } = newTestResult;
     const isStartAfterEnd = testTypes.some((testType) => {
       const { testTypeStartTimestamp, testTypeEndTimestamp } = testType;
       return DateProvider.isAfterDate(
-        testTypeStartTimestamp,
-        testTypeEndTimestamp,
+        testTypeStartTimestamp as string,
```

Comment:
surely if we correctly typed testType these would not need casted? I assume we have a 'testType' type.

Reply:
In the old models they were `any` type, so there were no issues going into the method below, however now we are typing correctly, these dates can be `string` or `null` which I think is why the `as string` was added.

```
  public static isAfterDate(
    date1: string | number | Date,
    date2: string | number | Date,
  ) {
    return moment(date1).isAfter(date2);
  }
```

So the code above would mean that if either of those values did happen to be null (which is possible according to some of the validators), then I think this method could fall over and throw an error. Which is consistent with the behaviour in the existing logic (where the type was `any`) but is probably still something we want to look at.